src/AvaloniaEdit/Editing/SelectionMouseHandler.cs

```diff
@@ -392,8 +392,15 @@ private void TextArea_OptionChanged(object sender, PropertyChangedEventArgs e)
 
         private void TextArea_MouseLeftButtonDown(object sender, PointerPressedEventArgs e)
         {
-            if (e.GetCurrentPoint(TextArea).Properties.IsLeftButtonPressed)
-            { 
+            if (e.GetCurrentPoint(TextArea).Properties.IsLeftButtonPressed == false)
+            {
+                if (TextArea.IsRightClickMovesCaret == true)
+                {
+                    SetCaretOffsetToMousePosition(e);
+                }
```

Comment:
should we ignore that if PointerPressedEventArgs is handled already?

Reply:
Probably doesn't make much difference in this case, but can't hurt.